src/Stream/src/StreamBase/StreamsHost/StreamsHost.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Steeltoe.Common.Contexts;
+using Steeltoe.Common.Lifecycle;
+using Steeltoe.Integration.Extensions;
+using Steeltoe.Stream.Binding;
+using Steeltoe.Stream.Extensions;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Steeltoe.Stream.StreamsHost
+{
+    public sealed class StreamsHost : IHost
+    {
+        public static IHostBuilder CreateDefaultBuilder<T>() => new StreamsHostBuilder<T>(Host.CreateDefaultBuilder());
+
+        public static IHostBuilder CreateDefaultBuilder<T>(string[] args) => new StreamsHostBuilder<T>(Host.CreateDefaultBuilder(args));
+
+        public StreamsHost(IHost host)
+        {
+            _host = host;
+        }
+
+        public IServiceProvider Services => _host.Services;
+
+        private readonly IHost _host;
+
+        public void Dispose()
+        {
+            _host.Dispose();
+        }
+
+        public Task StartAsync(CancellationToken cancellationToken = default)
+        {
+            var lifecycleProcessor = _host.Services.GetRequiredService<ILifecycleProcessor>();
+            lifecycleProcessor.OnRefresh();
+            var processor = _host.Services.GetRequiredService<StreamListenerAttributeProcessor>();
+            processor.AfterSingletonsInstantiated();
+            return _host.StartAsync(cancellationToken);
+        }
+
+        public Task StopAsync(CancellationToken cancellationToken = default)
+        {
+            var lifecycleProcessor = _host.Services.GetRequiredService<ILifecycleProcessor>();
+            lifecycleProcessor.Stop();
+
+            // Stop that thing
+            return _host.StopAsync(cancellationToken);
+        }
+    }
+
+    public class StreamsHostBuilder<T> : IHostBuilder
+    {
+        private readonly IHostBuilder _hostBuilder;
+
+        public StreamsHostBuilder(IHostBuilder hostBuilder)
+        {
+            _hostBuilder = hostBuilder.ConfigureServices(services =>
```

Comment:
This looks really close to `HostBuilderExtensions.AddStreamsServices<T>` - would it make sense to refactor?